server/src/models/people/populate.js

```diff
@@ -1,46 +1,181 @@
 
-import { people_years } from "../constants.js";
 import { get_standard_csv_file_rows } from "../load_utils.js";
 
-import {
-  camel_case_headcount_model_names,
-  snake_case_headcount_model_names,
-} from "./headcount_model_utils.js";
-
-const headcount_csv_names_by_headcount_model_name = _.chain(
-  camel_case_headcount_model_names
-)
-  .zip(snake_case_headcount_model_names)
-  .fromPairs()
-  .value();
-
-const format_csv_records_then_register_to_headcount_models = (
-  headcount_models_and_records
-) => {
-  _.each(headcount_models_and_records, (model_and_record) => {
-    _.each(model_and_record.records, (record) =>
-      _.each(
-        people_years,
-        (col) => (record[col] = record[col] && parseFloat(record[col]))
-      )
-    );
-    _.each(model_and_record.records, (record) =>
-      model_and_record.model.register(record)
-    );
-  });
-};
-
-export default function ({ models }) {
-  const headcount_models_and_records = _.map(
-    headcount_csv_names_by_headcount_model_name,
-    (headcount_csv_name, headcount_model_name) => ({
-      model: models[headcount_model_name],
-      records: get_standard_csv_file_rows(`${headcount_csv_name}.csv`),
-    })
+export default async function ({ models }) {
+  const { OrgEmployeeSummary, GovEmployeeSummary } = models;
+
+  const igoc_rows = get_standard_csv_file_rows("igoc.csv");
+
+  const process_employee_csv = (csv_name) =>
+    _.chain(get_standard_csv_file_rows(csv_name))
+      .reject(["dept_code", "ZGOC"])
+      .map(({ dept_code, dimension, ...data_columns }) => ({
+        dept_code,
+        dimension,
+        ..._.mapValues(data_columns, _.toNumber),
+      }))
+      .groupBy("dept_code")
+      .map((dept_rows, dept_code) => ({
+        org_id: _.find(
+          igoc_rows,
+          (igoc_row) => igoc_row.dept_code === dept_code
+        ).org_id,
+        data: _.chain(dept_rows)
+          .map(({ dept_code, dimension, avg_share, ...values_by_year }) => ({
+            dimension,
+            by_year: _.map(values_by_year, (value, year) => ({
+              year,
+              value,
+            })),
+            avg_share,
+          }))
+          .value(),
+      }))
+      .value();
+
+  const process_employee_data_sums = (csv_name) => {
+    const raw_data = get_standard_csv_file_rows(csv_name);
+    const years = _.chain(raw_data[0])
+      .keys()
+      .reject((key) => {
+        return isNaN(key);
+      })
+      .value();
+
+    return _.chain(raw_data)
+      .groupBy("dimension")
+      .map((dimension_arr, dimension_group) => ({
+        id: dimension_group,
+        dimension: dimension_group,
+        data: {
+          by_year: _.chain(years)
+            .map((year) => [
+              year,
+              _.sumBy(dimension_arr, (row) => _.toNumber(row[year])),
+            ])
+            .fromPairs()
+            .map((value, key) => ({
+              year: [key][0],
+              value: value,
+            }))
+            .value(),
```

Comment:
Could use a reduce, something like 
```
_.reduce(
  dimension_arr, // dimension_rows?
  (row, memo) => _.map(memo, ({year, value}) => ({year, value: value + row[year]}),
  _.map(years, (year) => ({year, value: 0}))
)
```

Map, filter, and reduce are something like the three primitives of functional data manipulation, can do most things in terms of them. The most simple use of a reduce is probably a sum. You don't _have_ to write everything in terms of them (lodash provides all its other utils to save us the time of re-implementing common patterns with them), but using them directly can help simplify things.